core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -90,14 +113,20 @@ public void onDriverInit(@NonNull DriverContext driverContext) {
   @NonNull
   @Override
   public CompletionStage<Boolean> reload() {
-    CompletableFuture<Boolean> result = new CompletableFuture<>();
-    RunOrSchedule.on(singleThreaded.adminExecutor, () -> singleThreaded.reload(result));
-    return result;
+    if (supportsReloading()) {
+      CompletableFuture<Boolean> result = new CompletableFuture<>();
+      RunOrSchedule.on(singleThreaded.adminExecutor, () -> singleThreaded.reload(result));
+      return result;
+    } else {
+      return CompletableFutures.failedFuture(
+          new UnsupportedOperationException(
+              "This instance of DefaultDriverConfigLoader does not support reloading"));
+    }
   }
 
   @Override
   public boolean supportsReloading() {
-    return true;
+    return supportsReloading;
```

Comment:
I see a potential problem when user will override `supportsReloading` and also will use this constructor:
`public DefaultDriverConfigLoader(Supplier<Config> configSupplier, boolean supportsReloading) {`
Shouldn't we make this method final to prevent overriding?

Reply:
We could, but at some point we cannot prevent users from doing all kinds of stupid things. My problem with final methods is that they cannot be mocked, so making it final could cause more harm than good eventually.

Reply:
Yes, but it seems a bit misleading to me. 
In a case when the user will call the parameterless constructor or `public DefaultDriverConfigLoader(Supplier<Config> configSupplier)` the local field of the class `private final boolean supportsReloading;` will have the value `true` whereas `supportsReloading()` may be overridden and returns `false`.
For me, it looks like the object will be in the inconsistent state - value says one thing, the method other.

Reply:
Ok ok, let's make those methods final.